jdbc-adapter/virtualschema-jdbc-adapter/src/test/java/com/exasol/adapter/dialects/FileBasedIntegrationTest.java

```diff
@@ -4,76 +4,78 @@
- * <ul>
- * <li>testQuery: A single string containing the test query. This test does not use the test query, since the testcase
- * parses the pushdown request directly.
- * <li>expectedPushdownRequest: A list of pushdownRequests as they are generated by the database. This is a list because
- * a single query can generate multiple pushdowns (e.g. join).
- * <li>expectedPushdownResponse: For each dialect that should be tested a list of strings with the returned Pushdown
- * SQLs.
+ * Writing a new test means writing new test data files. The test files are in JSON format and have to have the
+ * extension <code>.json</code>. The following attributes have to be present:
+ * <dl>
+ * <dt><code>testSchema</code></dt>
+ * <dd>This is the schema definition. This test does not use the schema definition, since the test case parses the
+ * pushdown request directly.</dd>
+ * <dt><code>testCases</code></dt>
+ * <dd>A list of test cases to be performed on the schema.
+ * <dd>
+ * </dl>
+ * <p>
+ * The list of test cases contains:
+ * </p>
+ * <dl>
+ * <dt><code>testQuery</code></dt>
+ * <dd>A single string containing the test query. This test does not use the test query, since the test case parses the
+ * pushdown request directly.</dd>
+ * <dt><code>expectedPushdownRequest</code></dt>
+ * <dd>A list of pushdownRequests as they are generated by the database. This is a list because a single query can
+ * generate multiple push-downs (e.g. <code>JOIN</code>).</dd>
+ * <dt><code>expectedPushdownResponse</code></dt>
+ * <dd>For each dialect that should be tested a list of strings with the returned push-down SQLs.</dd></dd>
+ * </dl>
  */
-@RunWith(Parameterized.class)
-public class FileBasedIntegrationTest {
+class FileBasedIntegrationTest {
     private static final String INTEGRATION_TESTFILES_DIR = "target/test-classes/integration";
     private static final String TEST_FILE_KEY_TESTCASES = "testCases";
     private static final String TEST_FILE_KEY_EXP_PD_REQUEST = "expectedPushdownRequest";
     private static final String TEST_FILE_KEY_EXP_PD_RESPONSE = "expectedPushdownResponse";
 
-    @Parameterized.Parameters(name = "{index}: {0}")
-    public static Iterable<? extends Object> data() {
+    static Stream<File> listTestFiles() {
         final File testDir = new File(INTEGRATION_TESTFILES_DIR);
-        return Arrays.asList(testDir.listFiles((dir, name) -> name.endsWith(".json")));
-    }
-
-    private final File testFile;
-
-    public FileBasedIntegrationTest(final File testFile) {
-        this.testFile = testFile;
+        return Arrays.stream(testDir.listFiles((dir, name) -> name.endsWith(".json")));
     }
 
-    @Test
-    public void testPushdownFromTestFile() throws Exception {
-        final String jsonTest = Files.toString(this.testFile, Charsets.UTF_8);
+    @MethodSource("listTestFiles")
+    @ParameterizedTest
+    void testPushdownFromTestFile(final File testFile) throws Exception {
```

Comment:
I think it's better to check and replace Exception with specific exceptions which are thrown

Reply:
Done.